lib/CIndexStoreDB/CIndexStoreDB.cpp

```diff
@@ -598,4 +599,32 @@ indexstoredb_index_includes_of_unit(
   });
 }
 
+bool
+indexstoredb_index_unit_tests_referenced_by_main_files(
+  _Nonnull indexstoredb_index_t index,
+   const char *_Nonnull const *_Nonnull cMainFilePaths,
+   size_t count,
+  _Nonnull indexstoredb_symbol_occurrence_receiver_t receiver
+) {
+  auto obj = (Object<std::shared_ptr<IndexSystem>> *)index;
+  SmallVector<CanonicalFilePathRef, 2> mainFilePaths;
+  for (size_t i = 0; i < count; ++i) {
+    mainFilePaths.push_back(CanonicalFilePathRef::getAsCanonicalPath(cMainFilePaths[i]));
```

Comment:
Should we *actually* canonicalize these? indexstoredb has a cache of canonicalized paths, so maybe better to do it here than in LSP.

Reply:
Done. A little bit annoying that we can only canonicalize at the `IndexSystemImpl` level (because it has the `PathCache`), so we need to create another copy of the array there.